src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java

```diff
@@ -0,0 +1,172 @@
+package io.kestra.plugin.notifications.discord;
+
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.VoidOutput;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.JacksonMapper;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+
+import java.awt.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public abstract class DiscordTemplate extends DiscordIncomingWebhook {
+
+    @Schema(
+        title = "Template to use",
+        hidden = true
+    )
+    @PluginProperty(dynamic = true)
+    protected String templateUri;
+
+    @Schema(
+        title = "Map of variables to use for the message template"
+    )
+    @PluginProperty(dynamic = true)
+    protected Map<String, Object> templateRenderMap;
+
+    @Schema(
+        title = "Webhook username who sends"
+    )
+    @PluginProperty(dynamic = true)
+    protected String username;
+
+    @Schema(
+        title = "Webhook profile photo who sends"
```

Comment:
```suggestion
        title = "Webhook profile photo URL"
```

Reply:
This is named avatar_url on the webhook property, please use the same property name on the task than on the Discord API to facilitate the task usage.